src/WinRT.Runtime/Marshalers.cs

```diff
@@ -25,55 +26,105 @@ public static void Dispose(this GCHandle handle)
                 handle.Free();
             }
         }
+
+#if !NET
+        public static unsafe ref readonly char GetPinnableReference(this string str)
+        {
+            return ref *(char*)((*(byte**)Unsafe.AsPointer(ref str)) + RuntimeHelpers.OffsetToStringData);

```

Comment:
This has GC hole that will lead to intermittent crashes during GC. It is not valid to convert between `ref` and unmanaged pointers like this.

Reply:
is there an easy way to bridge the gap for netstandard support?  I was hoping to isolate to a GetPinnableReference extension method.

Reply:
```
public static unsafe ref readonly char GetPinnableReference(this string str)
{
    fixed (char * p = str) { return ref *p; }
}
```

Reply:
This may work, assuming that the runtime has no bugs.